test/tstDetailsAlgorithms.cpp

```diff
@@ -9,47 +9,61 @@
  * SPDX-License-Identifier: BSD-3-Clause                                    *
  ****************************************************************************/
 
+#include <ArborX_Box.hpp>
 #include <ArborX_DetailsAlgorithms.hpp>
 #include <ArborX_DetailsKokkosExtArithmeticTraits.hpp>
+#include <ArborX_HyperBox.hpp>
+#include <ArborX_HyperPoint.hpp>
+#include <ArborX_HyperSphere.hpp>
+#include <ArborX_Point.hpp>
+#include <ArborX_Sphere.hpp>
 
 #define BOOST_TEST_MODULE Geometry
 #include <boost/test/unit_test.hpp>
 
 // NOTE: message is not required anymore with C++17
 #define STATIC_ASSERT(cond) static_assert(cond, "")
 
-using ArborX::Box;
-using ArborX::Point;
-using ArborX::Sphere;
+using GeometryBundles = std::tuple<
+    std::tuple<ArborX::Point, ArborX::Box, ArborX::Sphere>,
+    std::tuple<ArborX::Experimental::Point<3>, ArborX::Experimental::Box<3>,
+               ArborX::Experimental::Sphere<3>>>;
 
-BOOST_AUTO_TEST_CASE(distance)
+BOOST_AUTO_TEST_CASE_TEMPLATE(distance, Geometries_t, GeometryBundles)
```

Comment:
This way allows us to reuse existing 3D tests to test new geometries too, at least for dimension 3.